internal/controllers/bmc_reconciler.go

```diff
@@ -0,0 +1,152 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	apimcfgv1 "github.com/openshift/api/machineconfiguration/v1"
+	apioperatorv1 "github.com/openshift/api/operator/v1"
+	kmmv1beta1 "github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/constants"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/mcfg"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const BMCReconcilerName = "BootModuleConfigReconciler"
+
+// BMC reconciler handle the changes/creation of the MachineConfig that can load day1 kernel module
+type bmcReconciler struct {
+	reconHelper bmcReconcilerHelperAPI
+}
+
+func NewBMCReconciler(client client.Client, mcfgAPI mcfg.MCFG, scheme *runtime.Scheme) *bmcReconciler {
+	helper := newBMCReconcilerHelper(client, mcfgAPI, scheme)
+	return &bmcReconciler{
+		reconHelper: helper,
+	}
+}
+
+func (r *bmcReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&kmmv1beta1.BootModuleConfig{}).
+		Named(BMCReconcilerName).
+		Owns(&apimcfgv1.MachineConfig{}).
```

Comment:
Does it also need to watch `MachineConfiguration`? What happens if someone is manually modifying the machineConfiguration?